qiskit/optimization/problems/quadratic_program.py

```diff
@@ -1093,6 +1093,62 @@ def from_ising(self,
         offset -= offset
 
 
+    def is_feasible(self, x: Union[List[float], np.ndarray], detailed = False) -> bool:
+
+        satisfied_variables = []
+        for i, v in enumerate(x):    
+            variable = self.get_variable(i)
+            if variable._lowerbound <= v <= variable._upperbound :
+                satisfied_variables.append(True)
+                # print(f'{variable._name} is within the bounds') 
+            else :
+                satisfied_variables.append(False)
+                # print(f'{variable._name} is outside the bounds')
+
+        satisfied_linear_constraints = []
+        for constraint in self._linear_constraints:
+            lhs = constraint.evaluate(x)
+            if constraint.sense == ConstraintSense.LE:
+                satisfied_linear_constraints.append(lhs <= constraint.rhs)
+            elif constraint.sense == ConstraintSense.GE:
+                satisfied_linear_constraints.append(lhs >= constraint.rhs)
+            elif constraint.sense == ConstraintSense.EQ:
+                satisfied_linear_constraints.append(isclose(lhs, constraint.rhs))
+            else:
+                raise QiskitOptimizationError("Invalid sense!")
+
+        satisfied_quadratic_constraints = []    
+        for constraint in self._quadratic_constraints:
+            lhs = constraint.evaluate(x)
+            if constraint.sense == ConstraintSense.LE:
+                satisfied_quadratic_constraints.append(lhs <= constraint.rhs)
+            elif constraint.sense == ConstraintSense.GE:
+                satisfied_quadratic_constraints.append(lhs >= constraint.rhs)
+            elif constraint.sense == ConstraintSense.EQ:
+                satisfied_quadratic_constraints.append(isclose(lhs, constraint.rhs))
+            else:
+                raise QiskitOptimizationError("Invalid sense!")
+            
+            
+        if(detailed == True):
+
+            for i in range(len(satisfied_variables)):
+                (print(f'{self.get_variable(i)._name} is within the bounds') if satisfied_variables[i]
+                else logger.warning(f'{self.get_variable(i)._name} is outside the bounds'))
+
```

Comment:
The initial idea was to return a list of violations, not just printing them.

Reply:
> 
> 
> The initial idea was to return a list of violations, not just printing them.

This is for debug purposes only. I was not sure about how the return value was required. I will change it into a list.

Reply:
This is a good question how to return them back. May be we should postpone this feature, I mean returning a list of violations and focus on the boolean result.

Reply:
If it returns the violations as well, then we have to change the return type from ->bool to ->Tuple(bool, List[float]). Would that be alright?

Reply:
I'd expect to see a list of something, but what is float here?

Reply:
So currently I will print the violations and just return the bool result. One way would be to change the return type from ->bool to ->Tuple[bool, List[float]] and return empty list whenever detailed = False.

Reply:
The list can contain float values as well. Like x = [1.0, 2.0, 1.0, 1.99999982]

Reply:
Yeah, but what do they mean, these float values?

Reply:
I am sorry I didn't get your question. I would try to explain from what I understood. 
So if we define a quadratic problem with variables x, y and z, then we get results.x = [1.0, 2.0, 1.0] would mean x = 1.0, y = 2.0 and z = 1.0.